src/ucp/core/ucp_mm.inl

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_MM_INL_
+#define UCP_MM_INL_
+
+#include "ucp_mm.h"
+
+
+extern ucp_mem_h ucp_mem_dummy_handle;
+
+static UCS_F_ALWAYS_INLINE int
+ucp_memh_is_zero_length(const ucp_mem_h memh) {
+    return memh == ucp_mem_dummy_handle;
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_memh_get(ucp_context_h context, void *address, size_t length,
+             ucs_memory_type_t mem_type, ucp_md_map_t reg_md_map,
+             unsigned uct_flags, ucp_mem_h *memh_p)
+{
+    ucs_rcache_region_t *rregion;
+    ucs_status_t status;
+    ucp_mem_h memh;
+
+    if (length == 0) {
+        *memh_p = ucp_mem_dummy_handle;
+        return UCS_OK;
+    }
+
+    if (context->rcache != NULL) {
+        status = ucs_rcache_get(context->rcache, address, length,
+                                PROT_READ|PROT_WRITE, NULL, &rregion);
+        if (status != UCS_OK) {
+            return status;
+        }
+
+        memh = ucs_derived_of(rregion, ucp_mem_t);
+
+        if (ucs_test_all_flags(memh->md_map, reg_md_map)) {
+            *memh_p = memh;
+            return UCS_OK;
+        }
+
+        if (context->config.ext.reg_whole_alloc_bitmap & UCS_BIT(mem_type)) {
```

Comment:
1. pls documet why
2. can we push this down to ucp_memh_get_slow - if md_map is incomplete, put the region and fallback to slowpath